cmd/controller/disk.go

```diff
@@ -28,12 +28,16 @@ type DiskInfo struct {
 	Serial           string          // Serial is serial no of a disk
 	Vendor           string          // Vendor is vendor of a disk
 	Path             string          // Path is dev path of a disk like /dev/sda
+	FirmwareRevision string          // FirmwareRevision is the firmare revision for a disk
+	LogicalSize      uint32          // LogicalSize is the Logical size of a disk
+	Version          string          // Version is the disk version
 }
 
 // ProbeIdentifiers contains some keys to uniquely identify each disk by probe
 type ProbeIdentifier struct {
-	Uuid           string // Uuid is uid of disk which is generated by udev probe.
-	UdevIdentifier string // UdevIdentifier(syspath) used to identify disk by udevprobe.
+	Uuid            string // Uuid is uid of disk which is generated by udev probe.
+	UdevIdentifier  string // UdevIdentifier(syspath) used to identify disk by udevprobe.
+	SmartIdentifier string // SmartIdentifier (devPath) is used to identify disk by smartprobe.
```

Comment:
If the "smart identifier" is the devpath, please then call it devpath? These things have nothing to do directly with SMART, agree?

Reply:
Right @gila . But the naming convention we are following here is based on the probes such as for udev probe, it is udevIdentifier and for smart it is smartIdentifier.
And the underlying struct which defines this is like 

// Identifier (devPath such as /dev/sda,etc) is an identifier for smart probe
type Identifier struct {
	DevPath string
}